rethinkdb/datadog_checks/rethinkdb/check.py

```diff
@@ -0,0 +1,108 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from contextlib import contextmanager
+from typing import Any, Callable, Iterator, Sequence, cast
+
+import rethinkdb
+
+from datadog_checks.base import AgentCheck
+
+from . import operations, queries
+from .config import Config
+from .document_db import DocumentQuery
+from .document_db.types import Metric
+from .types import Instance
+
+
+class RethinkDBCheck(AgentCheck):
+    """
+    Collect metrics from a RethinkDB cluster.
+    """
+
+    SERVICE_CHECK_CONNECT = 'rethinkdb.can_connect'
+
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Any) -> None
+        super(RethinkDBCheck, self).__init__(*args, **kwargs)
+        self.config = Config(cast(Instance, self.instance))
+        self.queries = (
+            queries.config_summary,
+            queries.cluster_statistics,
+            queries.server_statistics,
+            queries.table_statistics,
+            queries.replica_statistics,
+            queries.table_statuses,
+            queries.server_statuses,
+            queries.system_jobs,
+            queries.current_issues_summary,
+        )  # type: Sequence[DocumentQuery]
+
+    @contextmanager
+    def connect_submitting_service_checks(self):
+        # type: () -> Iterator[rethinkdb.net.Connection]
+        config = self.config
+
+        tags = ['host:{}'.format(config.host), 'port:{}'.format(config.port)]
+        tags.extend(config.tags)
```

Comment:
Wdyta saving that since it's static?

Reply:
Reusing the list feels like micro-opt (with a risk of mutating it someday…), but I can at least define it as a `Config` property. :-)